pkgs/applications/graphics/ktikz/default.nix

```diff
@@ -0,0 +1,74 @@
+{ withKDE ? true,
+stdenv, fetchurl, qt, kdelibs, gettext, pkgconfig, poppler_qt4, auctex }:
+
+assert withKDE -> kdelibs != null;
+assert ! withKDE -> pkgconfig != null;
+
+let
+  version = "0.10";
+  commonBuildInputs = [ gettext qt poppler_qt4 auctex ];
+
+  qtikz = {
+    name = "qtikz-${version}";
+
+    conf = ''
+      # installation prefix:
+      #PREFIX = ""
+
+      # install desktop file here (*nix only):
+      DESKTOPDIR = ''$''${PREFIX}/share/applications
+
+      # install mimetype here:
+      MIMEDIR = ''$''${PREFIX}/share/mime/packages
+
+      CONFIG -= debug
+      CONFIG += release
+
+      # qmake command:
+      QMAKECOMMAND = qmake
+      # lrelease command:
+      LRELEASECOMMAND = lrelease
+      # qcollectiongenerator command:
+      #QCOLLECTIONGENERATORCOMMAND = qcollectiongenerator
+
+      # TikZ documentation default file path:
+      TIKZ_DOCUMENTATION_DEFAULT = ''$''${PREFIX}/share/doc/texmf/pgf/pgfmanual.pdf.gz
+    '';
+
+    patchPhase = ''
+      echo "$conf" > conf.pri
+    '';
+
+    configurePhase = ''
+      qmake PREFIX="$out" ./qtikz.pro
+    '';
+
+    buildInputs = commonBuildInputs ++ [ pkgconfig ];
+  };
+
+  ktikz = {
+    name = "ktikz-${version}";
+    buildInputs = commonBuildInputs ++ [ kdelibs ];
+  };
+
+  common = {
+    inherit version;
+    src = fetchurl {
+      url = "http://www.hackenberger.at/ktikz/ktikz_${version}.tar.gz";
```

Comment:
As per the blog [1]: 

> For Linux there are two versions available: KtikZ and QtikZ. KtikZ is integrated with the KDE4 platform, whereas QtikZ has the traditional Qt-only interface. 

[1] http://www.hackenberger.at/blog/ktikz-editor-for-the-tikz-language/